### PR TITLE
Add WFS permalink service

### PR DESCRIPTION
**Pull request by tsauerwein**
Adds a `ngeo.WfsPermalink` service which is used by `gmf.Permalink`. When calling the new example with e.g. `?wfs_layer=fuel&wfs_osm_id=1420918679`, a WFS GetFeature request is made and the features are shown with the `gmf-mobiledisplayqueries` directive.

For the yet to implement [WFS queries](https://github.com/camptocamp/c2cgeoportal/wiki/Spec-%231771-Querying-data-%28click-and-box%29), some functionality of `ngeo.WfsPermalink` might be refactored to avoid duplicated code.

Closes https://github.com/camptocamp/c2cgeoportal/issues/1670

### CODE REVIEW

File: src/services/wfspermalink.js [hunk 1]
```diff
@@ -0,0 +1,304 @@
+goog.provide('ngeo.WfsPermalink');
+
+goog.require('ngeo');
+/** @suppress {extraRequire} - required for `ngeoQueryResult` */
+goog.require('ngeo.Query');
+goog.require('ol.format.WFS');
+
```
Comment (fredj) [inline src/services/wfspermalink.js]: add `goog.require('ol.format.ogc.filter');`?

Comment (tsauerwein) [inline src/services/wfspermalink.js]: mhm no, doesn't fix the error. Do I maybe need to add `@export` to the namespace in OpenLayers?

Comment (tsauerwein) [inline src/services/wfspermalink.js]: On the other hand, I worked already when I was working with my own ol3 branch. After that I switched to ol3 master. I will investigate next week.

Comment (fredj) [inline src/services/wfspermalink.js]: or maybe it's the cache (node_modules) in travis ?

Comment (tsauerwein) [inline src/services/wfspermalink.js]: It also fails locally for me, and I already tried `make cleanall`.

File: test/spec/services/wfspermalink.spec.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
+goog.require('ngeo.WfsPermalink');
+goog.require(' ol.format.ogc.filter');
```
Comment (fredj) [inline test/spec/services/wfspermalink.spec.js]: extra space !

Comment (tsauerwein) [inline test/spec/services/wfspermalink.spec.js]: Great, nice find Fred! We should run the linter on the test files.

Comment (tsauerwein) [inline test/spec/services/wfspermalink.spec.js]: This fixed the error, now I have another problem related to a rebase. But I will look into that next week.

File: test/spec/services/query.spec.js [hunk 3]
```diff
@@ -72,8 +72,8 @@ describe('ngeo.Query', function() {
 
       var url = 'https://geomapfish-demo.camptocamp.net/1.6/wsgi/mapserv_proxy';
 
-      var requestUrlBusStop = url + '?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetFeatureInfo&FORMAT=image%2Fpng&TRANSPARENT=true&INFO_FORMAT=application%2Fvnd.ogc.gml&FEATURE_COUNT=50&I=50&J=50&CRS=EPSG%3A21781&STYLES=&WIDTH=101&HEIGHT=101&BBOX=489100%2C119900.00000000003%2C509300%2C140100.00000000003&LAYERS=bus_stop&QUERY_LAYERS=bus_stop';
-      var requestUrlBusStopAndInformation = url + '?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetFeatureInfo&FORMAT=image%2Fpng&TRANSPARENT=true&INFO_FORMAT=application%2Fvnd.ogc.gml&FEATURE_COUNT=50&I=50&J=50&CRS=EPSG%3A21781&STYLES=&WIDTH=101&HEIGHT=101&BBOX=523700%2C142900.00000000003%2C543900%2C163100.00000000003&LAYERS=information%2Cbus_stop&QUERY_LAYERS=information%2Cbus_stop';
+      var requestUrlBusStop = url + '?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetFeatureInfo&FORMAT=image%2Fpng&TRANSPARENT=true&INFO_FORMAT=application%2Fvnd.ogc.gml&FEATURE_COUNT=50&I=50&J=50&CRS=EPSG%3A21781&STYLES&WIDTH=101&HEIGHT=101&BBOX=489100%2C119900.00000000003%2C509300%2C140100.00000000003&LAYERS=bus_stop&QUERY_LAYERS=bus_stop';
```
Comment (tsauerwein) [inline test/spec/services/query.spec.js]: Instead of `&STYLES=&WIDTH=101` it is now `&STYLES&WIDTH=101` (without `=` for `styles`).

**Comment by tsauerwein** [comment]
Ready to review

File: contribs/gmf/examples/wfs-permalink.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+/** @const **/
+var app = {};
+
+
+/** @type {!angular.Module} **/
+app.module = angular.module('app', ['gmf']);
+
+app.module.constant('ngeoWfsPermalinkOptions',
+    /** @type {ngeox.WfsPermalinkOptions} */ ({
+      url: 'https://geomapfish-demo.camptocamp.net/1.6/wsgi/mapserv_proxy',
```
Comment (fredj) [inline contribs/gmf/examples/wfs-permalink.js]: change to `https://geomapfish-demo.camptocamp.net/2.1/wsgi/mapserv_proxy`

File: test/spec/services/wfspermalink.spec.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+describe('ngeo.WfsPermalink', function() {
+
+  var ngeoWfsPermalink;
+  var ngeoQueryResult;
+
+  beforeEach(function() {
+    module('ngeo', function($provide) {
+      $provide.constant('ngeoWfsPermalinkOptions', {
+        url: 'https://geomapfish-demo.camptocamp.net/2.0/wsgi/mapserv_proxy',
```
Comment (fredj) [inline test/spec/services/wfspermalink.spec.js]: change to `https://geomapfish-demo.camptocamp.net/2.1/wsgi/mapserv_proxy`

File: test/spec/services/wfspermalink.spec.js [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    expect(ngeoWfsPermalink instanceof ngeo.WfsPermalink).toBe(true);
+  });
+
+  describe('#issue', function() {
+
+    var $httpBackend;
+    var map;
+
+    beforeEach(function() {
+      var url = 'https://geomapfish-demo.camptocamp.net/2.0/wsgi/mapserv_proxy';
```
Comment (fredj) [inline test/spec/services/wfspermalink.spec.js]: change to `https://geomapfish-demo.camptocamp.net/2.1/wsgi/mapserv_proxy`

**Comment by fredj** [comment]
LGTM

**Comment by tsauerwein** [comment]
Thanks for the review! I will merge when Travis passes.

**Pull request merged**